src/main/java/hudson/tasks/junit/TestResult.java

```diff
@@ -272,32 +268,42 @@ public TestResult(TestResult src) {
         this.impl = null;
     }
 
-    private static final XMLInputFactory factory = XMLInputFactory.newInstance();
+    static final XMLInputFactory xmlFactory = XMLInputFactory.newInstance();
+    static boolean USE_SAFE_XML_FACTORY =
+            Boolean.parseBoolean(System.getProperty(TestResult.class.getName() + ".USE_SAFE_XML_FACTORY","true"));
+    static boolean isXmlFactoryAdjusted = false;
+    
+    public static XMLInputFactory getXmlFactory() {
+        if (!isXmlFactoryAdjusted && USE_SAFE_XML_FACTORY) {
+            xmlFactory.setProperty(XMLInputFactory.SUPPORT_DTD, false);
+            xmlFactory.setProperty(XMLInputFactory.IS_REPLACING_ENTITY_REFERENCES, false);
+        }
+        return xmlFactory;
+    }
```

Comment:
this appears to not set isXMLFactoryAdjusted anywhere so it will aways set these properties.

However you do not need this - and you can move this all to a single static initialiser or helper method to get the XMLFactory.

e.g.

```java
 private static final XMLInputFactory factory;
 
 static {
     factory = XMLInputFactory.newInstance();
     if {SystemProperties.getBoolean(TestResult.class.getName() + ".USE_SAFE_XML_FACTORY", true)) {
        factory.setProperty(XMLInputFactory.SUPPORT_DTD, Boolean.FALSE);
        factory.setProperty(XMLInputFactory.IS_REPLACING_ENTITY_REFERENCES, Boolean.FALSE);    
 }

// if subclasses need access?
protected XMLInputFactory getXMLFactory() {
    return factory;
}
```

Reply:
Fixed.

Is there a reason to use Boolean.FALSE instead of false?

Reply:
The method takes an Object not a boolean.

By using a boolean primitive the code will rely on auto-boxing which should be avoided if you have a fixed value.

A modern compiler could be intelligent enough to inline this to avoid the auto-boxing, but it's still at the end of the day wanting an Object.